lib/datasource/nuget/v3.ts

```diff
@@ -29,97 +39,121 @@ export async function getQueryUrl(url: string): Promise<string | null> {
   }
 
   try {
-    // TODO: fix types
-    const servicesIndexRaw = await http.getJson<any>(url);
-    const searchQueryService = servicesIndexRaw.body.resources.find(
-      (resource) => resource['@type']?.startsWith(resourceType)
-    );
-    const searchQueryServiceId = searchQueryService['@id'];
-
+    const servicesIndexRaw = await http.getJson<ServicesIndexRaw>(url);
+    const services = servicesIndexRaw.body.resources
+      .filter((resource) => {
+        const [type, version] = resource['@type']?.split('/') || [];
+        return type === resourceType && semver.valid(version);
+      })
+      .sort((x, y) => {
+        const [, xver] = x['@type']?.split('/');
+        const [, yver] = y['@type']?.split('/');
+        return semver.compare(xver, yver);
+      });
```

Comment:
```suggestion
    const services = servicesIndexRaw.body.resources
      .map((resource) =>({ '@id': resource['@id'], type: resource['@type']?.split('/') || [] })
      .filter((resource) => {
        const [type, version] = resource.type;
        return type === resourceType && semver.valid(version);
      })
      .sort((x, y) => {
        const [, xver] = x.type;
        const [, yver] = y.type;
        return semver.compare(xver, yver);
      });
```

What about something like this?